### PR TITLE
[docs, ri2] Fixing broken autosectionheaders and intersphinx links

### LINKED ISSUE #2808: [docs] Enable interspinx mapping and autosectionheaders globally
Some sub-documents, such as RI2, started to use autosectionheaders and intersphinx mapping. These features however are not yet available when building the entire Anuket specs documentation, because they are not enabled in the global conf.py.

### PR DESCRIPTION
**Pull request by gkunz**
As part of the rst conversion, RI2 enabled autosectionheaders and
intersphinx mapping links. However, when building the entire Anuket
documentation, these links broke because autosectionheaders and
intersphinx mapping was not enabled globally. This change adds
intersphinx mapping and autosectionheaders to the global conf.py

Fixes #2808

Signed-off-by: Georg Kunz <georg.kunz@ericsson.com>

### CODE REVIEW

**Review by pgoyal01 — APPROVED** [review]

**Review by CsatariGergely — APPROVED** [review]

**Review by collivier — COMMENTED** [review]
I won't support the tox.ini modification. Else I would be fine with this commit.

File: doc/ref_impl/cntt-ri2/tox.ini [hunk 1]
```diff
@@ -3,7 +3,7 @@ envlist = docs
 skipsdist = True
 
 [testenv:docs]
-basepython = python3.9
+basepython = python3
```
[the remark this answers is not in the document]
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/tox.ini]: You are right about this being off-topic. I reverted this.

This tox.ini is for individual RI2 builds. I think the more generic definition like Python3 should for us - which is current being used in the global tox.ini.

**Review by rihabbanday — APPROVED** [review]

**Comment by rihabbanday** [comment]
> @rihabbanday can you please review? Thanks

Looks like @walterkozlowski must also approve it before it can be merged.

**Comment by gkunz** [comment]
@walterkozlowski can you please review. This fixes broken links in RI2. Thank you.

**Comment by gkunz** [comment]
@walterkozlowski kind reminder.

**Review by walterkozlowski — APPROVED** [review]

**Pull request merged**